cluster-scope/base/rbac.authorization.k8s.io/clusterrolebindings/cluster-reader/clusterrolebinding.yaml

```diff
@@ -0,0 +1,16 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+    annotations:
+        rbac.authorization.kubernetes.io/autoupdate: "true"
```

Comment:
https://kubernetes.io/docs/reference/access-authn-authz/rbac/#auto-reconciliation

> At each start-up, the API server updates default cluster roles with any missing permissions, and updates default cluster role bindings with any missing subjects. This allows the cluster to repair accidental modifications, and helps to keep roles and role bindings up-to-date as permissions and subjects change in new Kubernetes releases.

> To opt out of this reconciliation, set the rbac.authorization.kubernetes.io/autoupdate annotation on a default cluster role or rolebinding to false. Be aware that missing default permissions and subjects can result in non-functional clusters.

> Auto-reconciliation is enabled by default if the RBAC authorizer is active.
API discovery roles

Reply:
If setting it to true means that kubernetes will reset any modifications we make, then we might as well have a separate clusterrolebinding instead of setting autopudate to false which maybe risky. If I did have a separate clusterrolebinding then I could refer to the same clusterrole, right?

Reply:
Sure, that would work. I was tempted to try out this change first, and see if it "sticks" or not, but if you'd rather than just go with a new rolebinding that would be fine.

Reply:
I tested it on the staging cluster and I think it sticks, but it added another annotation to it which reports last-applied-configuration.

After I oc applied I got this
```
Warning: resource clusterrolebindings/cluster-readers is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by oc apply. oc apply should only be used on resources created declaratively by either oc create --save-config or oc apply. The missing annotation will be patched automatically.
clusterrolebinding.rbac.authorization.k8s.io/cluster-readers configured
```

and then the updated ClusterRoleBinding showed the new group
```
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"rbac.authorization.k8s.io/v1","kind":"ClusterRoleBinding","metadata":{"annotations":{"rbac.authorization.kubernetes.io/autoupdate":"true"},"creationTimestamp":"2021-11-01T22:33:55Z","name":"cluster-readers","resourceVersion":"11921","uid":"d995bcd8-2899-48c5-8235-40d9a222bdc7"},"roleRef":{"apiGroup":"rbac.authorization.k8s.io","kind":"ClusterRole","name":"cluster-reader"},"subjects":[{"apiGroup":"rbac.authorization.k8s.io","kind":"Group","name":"system:cluster-readers"},{"apiGroup":"rbac.authorization.k8s.io","kind":"Group","name":"lars-sandbox-member"}]}
    rbac.authorization.kubernetes.io/autoupdate: "true"
  creationTimestamp: "2021-11-01T22:33:55Z"
  name: cluster-readers
  resourceVersion: "355472076"
  uid: d995bcd8-2899-48c5-8235-40d9a222bdc7
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: cluster-reader
subjects:
- apiGroup: rbac.authorization.k8s.io
  kind: Group
  name: system:cluster-readers
- apiGroup: rbac.authorization.k8s.io
  kind: Group
  name: lars-sandbox-member
  ```

Reply:
That's okay; that's typical when using `oc apply` on something that was previously created using something other than `oc apply`.